### PR TITLE
Test: add `deneb` spec test version `beta.0`

### PR DESCRIPTION
**Pull request by terencechain**
prereq, we need to merge these first:
~https://github.com/prysmaticlabs/prysm/pull/12577~
~https://github.com/prysmaticlabs/prysm/pull/12565~

### CODE REVIEW

**Comment by prestonvanloon** [comment]
I think this needs to be rebased? Diff seems really large

**Comment by terencechain** [comment]
@potuz thanks for the feedback, I fixed them in [6f32e9b](https://github.com/prysmaticlabs/prysm/pull/12610/commits/6f32e9ba24238f10d9168d921e825fc9f9a77f6d)
one thing noting pointed out by you is we are missing the genesis folder, I will go ahead and create an issue to track that

File: testing/spectest/shared/deneb/ssz_static/ssz_static.go [hunk 1]
```diff
@@ -47,12 +49,14 @@ func unmarshalledSSZ(t *testing.T, serializedBytes []byte, folderName string) (i
 	case "AggregateAndProof":
 		obj = &ethpb.AggregateAttestationAndProof{}
 	case "BeaconBlock":
+		t.Skip("Skipping BeaconBlock test")
```
Comment (james-prysm) [inline testing/spectest/shared/deneb/ssz_static/ssz_static.go:52]: are we supposed to be skipping these tests? or is it taken cared of in a later pr

File: testing/spectest/shared/deneb/operations/withdrawals.go [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+			RunBlockOperationTest(t, folderPath, body, func(_ context.Context, s state.BeaconState, b interfaces.SignedBeaconBlock) (state.BeaconState, error) {
+				payload, err := b.Block().Body().Execution()
+				if err != nil {
+					return nil, err
+				}
+				withdrawals, err := payload.Withdrawals()
+				if err != nil {
+					return nil, err
+				}
+				p, err := consensusblocks.WrappedExecutionPayloadCapella(&enginev1.ExecutionPayloadCapella{Withdrawals: withdrawals}, 0)
```
Comment (james-prysm) [inline testing/spectest/shared/deneb/operations/withdrawals.go:43]: wait shouldn't this type be deneb?

Comment (terencechain) [reply testing/spectest/shared/deneb/operations/withdrawals.go]: fixed in 9f2d53754588ce54b10e297e8c765246964e069c

File: testing/spectest/shared/deneb/ssz_static/ssz_static.go (hunk 1 above)
Comment (terencechain) [reply testing/spectest/shared/deneb/ssz_static/ssz_static.go]: fixed in 9f2d53754588ce54b10e297e8c765246964e069c

**Review by james-prysm — APPROVED** [review]

**Pull request merged**